src/Raven.Studio/typescript/models/database/query/queryCriteria.ts

```diff
@@ -67,15 +67,22 @@ class queryCriteria {
             hash: genUtils.hashCode(name + (queryText || "")) } as storedQueryDto;
     }
 
-    setSelectedIndex(indexName: string) {
+    setSelectedIndex(indexName: string): void {
         let rql = "from ";
-        if (indexName.startsWith(queryUtil.DynamicPrefix)) {
+
+        if (indexName.toLocaleLowerCase().startsWith(queryUtil.AutoPrefix) && indexName.includes("'")) {
```

Comment:
please test this with query:

```
from "Orders" 
where 'Company""' = "companies/85-A"
```

query will be: `from index 'Auto/Orders/By'Company'And'Company""''`

Reply:
Should we handle that ?
When you create properties in a document you **cannot** create (at least from the studio)
properties like: `Company""` in the json document
i.e. 
`"Company""" : "some text"` is error

You **can** create properties with single quotes like: `Company''aaa`
i.e.
`"Company''aaa" : "some text"` is ok

Reply:
looks like we can:
```
{
    "LastAccessedUtc": "2021-11-21T07:58:32.9125919Z",
    "a\"'a": 2,
    "@metadata": {
        "@collection": "DemoStats",
        "Raven-Clr-Type": "DemoCommon.Models.DemoStats, DemoCommon"
    }
}
```

Anyway - its do defensive way and simply escape that w/o caring if we can or can't.

Reply:
I think we already have functions that escape that so let's just call it.

Reply:
done

Reply:
we have utility methods in queryUtils. You can even create `escapeIndexName` method and use anywhere we need to escape. Do you remember issue when index was named `index`? It is better to have such things in single place and in case of issue resolve it in single place.

Reply:
done